# Copyright (c) 2018 Kent R. Spillner <kspillner@acm.org>
#
# Permission to use, copy, modify, and distribute this software for any
# purpose with or without fee is hereby granted, provided that the above
# copyright notice and this permission notice appear in all copies.
#
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

Avoid creating and chmod'ing $srcdir unnecessarily

--- makepkg.orig	2017-12-18 17:27:49.000000000 -0600
+++ makepkg	2017-12-18 17:29:38.000000000 -0600
@@ -2344,8 +2344,12 @@
 fi
 
 # get back to our src directory so we can begin with sources
-mkdir -p "$srcdir"
-chmod a-s "$srcdir"
+if [[ ! -d "$srcdir" ]]; then
+	mkdir -p "$srcdir"
+fi
+if [[ -k "$srcdir" ]]; then
+	chmod a-s "$srcdir"
+fi
 cd_safe "$srcdir"
 
 if (( !REPKG )); then
